backend/routes/user.js

```diff
@@ -72,4 +139,49 @@ function deleteApiKey(req, res, next) {
 	}
 }
 
+async function getStarredMetadataTags(req, res, next) {
+	const username = req.session.user.username;
+	req.dataModel = await User.getStarredMetadataTags(username);
+	next();
+}
+
+async function appendStarredMetadataTag(req, res, next) {
+	const username = req.session.user.username;
+	const tag = req.body.tag;
+	if (!tag) {
+		res.status(400);
+		res.end();
+		return;
+	}
+
+	req.dataModel = await User.appendStarredMetadataTag(username,tag);
+	next();
+}
+
+async function replaceStarredMetadataTags(req, res, next) {
+	const username = req.session.user.username;
+	const tags = req.body;
+
+	if (!isArray(tags) || !tags.every(isString)) {
```

Comment:
Actually it would be the same thing. 
I need to know that every tag is efectivly a string. 
with some it would be 
tag.some(isNotString) 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

![image](https://user-images.githubusercontent.com/42032433/55085930-68d01400-509f-11e9-88f5-9d76e1546dc5.png)

lets say that the element 10 isNotString , then it would stop there , because is a truthy value, effectivily iterating 10 elements. 

In case that every element is actually a string, it would iterate ALL elements because it didnt find a truthy value


From: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

![image](https://user-images.githubusercontent.com/42032433/55085736-13940280-509f-11e9-827e-8b392cf430e5.png)

In case of let say the element 10 is not a string , then isString will return false, therefore stoping at the 10nth element. Iterating the same amount of elements as with some.

In case that every element is actuallu a strng it would iterata ALL elements just like .some would.

Reply:
Make sense! ;)